python/openassetio_traitgen/__init__.py

```diff
@@ -136,27 +135,34 @@ def generate(
     # As a convenience, log the parsed structure
     _log_package_declaration(package_declaration, logger)
 
-    if dry_run:
+    # Don't allow empty string to fall through as getattr will throw.
+    if not generator:
+        logger.error("No generator specified")
+        return
+
+    # Retrieve the generator by looking up an attribute with the
+    # requested name and generate.
+    generator_obj = getattr(generators, generator, None)
+    if generator_obj is None:
+        logger.error("Cannot find generator %s", generator)
         return
 
-    # Generate all requested languages
-    for language in languages:
+    if dry_run:
+        return
 
-        logger.info("Generating %s...", language)
-        child_logger = logger.getChild(language)
+    # Generate with requested generator
+    logger.info("Generating with %s...", generator)
```

Comment:
[consider][minor]  This will presumably say things like "Generating with python", "Generating with cpp", which sounds like it's the language used to generate, rather than the language being generated for. We could either revert this line or
```suggestion
    logger.info("Generating with generator '%s'...", generator)
```
to disambiguate.

Reply:
Yeah good point, i prefer your suggestion to reverting, will do that.

Reply:
[e466fea](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/7/commits/e466feae8fe9f8310bd83f1c7b12db47b85187dd)